src/sagas.js

```diff
@@ -304,15 +316,39 @@ function* refreshData (action) {
     console.log('refreshData(): not quiet')
     yield put(loadInProgress({ value: true }))
   }
-  yield put(getAllVms({ shallowFetch: !!action.payload.shallowFetch }))
-  yield put(getAllPools())
+  yield put(getVmsByCount({ count: action.payload.page * AppConfiguration.pageLimit, shallowFetch: !!action.payload.shallowFetch }))
```

Comment:
If user add small count of VMs it just fetch VMs and add new to list, if count will be bigger it turn on infinite scroll feature and start to fetching next pages until all VMs will be loaded.